### PR TITLE
Don't use empty files to indicate presence of tasty in class path.

### PR DESCRIPTION
**Pull request by DarkDimius**
Use hasTasty files instead.

### CODE REVIEW

**Comment by DarkDimius** [comment]
@smarter are IDE tests run by default?

**Comment by smarter** [comment]
@DarkDimius There are no IDE tests currently :).

File: compiler/src/dotty/tools/dotc/interactive/InteractiveDriver.scala [hunk 1]
```diff
@@ -86,10 +86,10 @@ class InteractiveDriver(settings: List[String]) extends Driver {
             else
               null
           prefix != null && {
-            val tastyFile = prefix + ".tasty"
+            val hasTastyFile = prefix + ".hasTasty"
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/interactive/InteractiveDriver.scala]: This won't work correctly if YemitTasty is enabled.

**Comment by smarter** [comment]
.tasty handling in sbt-dotty: https://github.com/lampepfl/dotty/blob/master/sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala#L100

File: sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala [hunk 2]
```diff
@@ -97,9 +97,8 @@ object DottyPlugin extends AutoPlugin {
       def delete(classes: Iterable[File]): Unit = {
         inherited.delete(classes flatMap { classFile =>
           val dottyFiles = if (classFile.getPath endsWith ".class") {
-            val f = new File(classFile.getAbsolutePath.stripSuffix(".class") + ".tasty")
-            if (f.exists) List(f)
-            else Nil
+            val prefix = classFile.getAbsolutePath.stripSuffix(".class")
+            List(".tasty", ".hasTasty").map(x => new File(prefix + x)).filter(_.exists)
```
Comment (smarter) [inline sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala]: Documentation above this method needs to be updated to talk about .hasTasty files too.

**Comment by DarkDimius** [comment]
ping @smarter. The doc was updated. Anything else? In absence of tests it's tricky to know)

**Comment by DarkDimius** [comment]
@smarter, why did you terminate the CI?

File: compiler/src/dotty/tools/backend/jvm/GenBCode.scala [hunk 3]
```diff
@@ -214,7 +214,8 @@ class GenBCodePipeline(val entryPoints: List[Symbol], val int: DottyBackendInter
             } else if (!outTastyFile.isVirtual) {
               // Create an empty file to signal that a tasty section exist in the corresponding .class
               // This is much cheaper and simpler to check than doing classfile parsing
-              outTastyFile.create()
+              val hasTasty = getFileForClassfile(outF, store.name, ".hasTasty")
```
Comment (smarter) [inline compiler/src/dotty/tools/backend/jvm/GenBCode.scala]: It turns out that `getFileForClassfile` will automatically create an empty file, so we are now creating both ".tasty" and ".hasTasty" files, I'll fix it :).

Comment (DarkDimius) [inline compiler/src/dotty/tools/backend/jvm/GenBCode.scala]: Thanks!

**Pull request merged**